### ISSUE TITLE
Support dual-stack in cni: canal

### ISSUE THREAD

**Issue by manuelbuil** [rancher/rke2#1883]
Is your feature request related to a problem? Please describe.

It is currently not possible to deploy canal using dual-stack automatically

Describe the solution you'd like

When configuring the initial cluster-cidr and service-cidr with ipv4 and ipv6 addresses, I'd like canal to reflect that and deploy a dual-stack networking infra

Additional context

**Comment by manuelbuil** [comment]
After investigating, this is not possible to do as canal does not support dual-stack. The problem is with the ipam configuration:
```
              "type": "host-local",
              "subnet": "usePodCidr"
```
in calico code, it uses the function https://github.com/projectcalico/cni-plugin/blob/v3.19.2/pkg/k8s/k8s.go#L894-L909 to fetch the ips and that one will only return one cidr. It should call getPodCidrs which will return a slice of cidrs. As the returned value is different, it will require some code changes

**Comment by manishmaharjan** [comment]
@manuelbuil , @cwayne18 , Is this still undergoing development or still in the blocked in Development at the moment? 
Any estimate for a release in any specific milestone? Thanks!

**Comment by manuelbuil** [comment]
> @manuelbuil , @cwayne18 , Is this still undergoing development or still in the blocked in Development at the moment? Any estimate for a release in any specific milestone? Thanks!

This is blocked by upstream canal: https://github.com/projectcalico/cni-plugin/issues/1177. Once canal supports dual-stack, we will be able to support it in rke2. In the meanwhile, you'll need to use another cni plugin

**Comment by manuelbuil** [comment]
Replaced by this new issue https://github.com/rancher/rke2/issues/2342
